include/deal.II/lac/sparse_matrix.templates.h

```diff
@@ -171,12 +171,22 @@ namespace internal
     typedef types::global_dof_index size_type;
 
     template <typename T>
-    void zero_subrange (const size_type begin,
-                        const size_type end,
-                        T *dst)
+    typename std::enable_if<std::is_trivial<T>::value>::type
+    zero_subrange (const size_type begin,
+                   const size_type end,
+                   T              *dst)
     {
       std::memset (dst+begin,0,(end-begin)*sizeof(T));
     }
+
+    template <typename T>
+    typename std::enable_if<!std::is_trivial<T>::value>::type
+    zero_subrange (const size_type begin,
+                   const size_type end,
+                   T              *dst)
+    {
+      std::fill (dst+begin,dst+end,0);
```

Comment:
I can think of types that do not allow the statement
```
  it = value;
```
with `value` of type `int` and value `0`, but it may be for another time to write this as `T(0)` or something else.